plugins/modules/file_remove.py

```diff
@@ -0,0 +1,297 @@
+                raise AssertionError(f"Failed to list directory {path}: {to_native(e)}")
+    else:
+        # Use glob pattern matching
+        if recursive:
+            glob_pattern = os.path.join(path, "**", pattern)
+            matching_files = [f for f in glob.glob(glob_pattern, recursive=True) if should_include_file(f, file_type)]
+        else:
+            glob_pattern = os.path.join(path, pattern)
+            matching_files = [f for f in glob.glob(glob_pattern) if should_include_file(f, file_type)]
+
+    return sorted(matching_files)
+
+
+def should_include_file(file_path, file_type):
+    """Determine if a file should be included based on its type."""
+    # Never include directories
+    if os.path.isdir(file_path):
+        return False
+
+    is_link = os.path.islink(file_path)
+    is_file = os.path.isfile(file_path)
+
+    if file_type == "file":
+        # Only regular files, not symlinks
+        return is_file and not is_link
+    elif file_type == "link":
+        # Only symbolic links
+        return is_link
+    elif file_type == "any":
+        # Both files and symlinks
+        return is_file or is_link
+
+    return False
+
+
+def remove_files(module, files):
+    """Remove the specified files and return results."""
+    removed_files = []
+    failed_files = []
+
+    for file_path in files:
+        try:
+            if module.check_mode:
+                # In check mode, just verify the file exists
+                if os.path.exists(file_path):
+                    removed_files.append(file_path)
+            else:
+                # Actually remove the file
+                os.remove(file_path)
+                removed_files.append(file_path)
+        except OSError as e:
+            failed_files.append((file_path, to_native(e)))
+
+    return removed_files, failed_files
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(type="path", required=True),
+            pattern=dict(type="str", required=True),
+            use_regex=dict(type="bool", default=False),
+            recursive=dict(type="bool", default=False),
+            file_type=dict(type="str", default="file", choices=["file", "link", "any"]),
+        ),
+        supports_check_mode=True,
+    )
+
+    path = module.params["path"]
+    pattern = module.params["pattern"]
+    use_regex = module.params["use_regex"]
+    recursive = module.params["recursive"]
+    file_type = module.params["file_type"]
+
+    # Validate that the path exists and is a directory
+    if not os.path.exists(path):
+        module.fail_json(msg=f"Path does not exist: {path}")
+
+    if not os.path.isdir(path):
+        module.fail_json(msg=f"Path is not a directory: {path}")
+
+    # Validate regex pattern if use_regex is true
+    if use_regex:
+        try:
+            re.compile(pattern)
+        except re.error as e:
+            module.fail_json(msg=f"Invalid regular expression pattern: {to_native(e)}")
```

Comment:
This is redundant, since we no longer support Python 2, more occurrences around the code should be adjusted.
```suggestion
            module.fail_json(msg=f"Invalid regular expression pattern: {e}")
```

Reply:
to_native import removed and all occurances changed.

163: to_native(e) - changed
214: to_native(e) - changed
255: to_native(e) - changed

from ansible.module_utils.common.text.converters import to_native - Removed